File: frontend/src/types/engineRegistry.ts
export interface ChangeLogEntry {
  date: string
  commit: string | null
  description: string
}

export interface ContractSpec {
  type: string
  fields: string[]
  notes: string | null
}

export interface EngineRecord {
  human_name: string
  technical_id: string
  engine_type: string
  lifecycle_status: string
  created_date: string | null
  frozen_date: string | null
  retired_date: string | null
  rulebook_status: string
  purpose: string
  key_characteristics: string[]
  input_contract: ContractSpec
  output_contract: ContractSpec
  dependencies: string[]
  supersedes: string | null
  superseded_by: string | null
  validation_status: string
  notes: string | null
  change_log: ChangeLogEntry[]
}

export interface EngineListResponse {
  engines: EngineRecord[]
  count: number
}

File: backend/api/schemas/engine_registry.py
"""
API schemas for the Research Engine Registry endpoints.

Read-only metadata visibility only.
No engine selection, no pipeline configuration, no mutation.
"""
from typing import Optional

from pydantic import BaseModel


class ChangeLogEntry(BaseModel):
    date: str
    commit: Optional[str] = None
    description: str


class ContractSpec(BaseModel):
    type: str
    fields: list[str]
    notes: Optional[str] = None


class EngineRecordResponse(BaseModel):
    human_name: str
    technical_id: str
    engine_type: str
    lifecycle_status: str
    created_date: Optional[str] = None
    frozen_date: Optional[str] = None
    retired_date: Optional[str] = None
    rulebook_status: str
    purpose: str
    key_characteristics: list[str]
    input_contract: ContractSpec
    output_contract: ContractSpec
    dependencies: list[str]
    supersedes: Optional[str] = None
    superseded_by: Optional[str] = None
    validation_status: str
    notes: Optional[str] = None
    change_log: list[ChangeLogEntry]


class EngineListResponse(BaseModel):
    engines: list[EngineRecordResponse]
    count: int
